src/modules/material-requests/services/material-requests.service.ts

```diff
@@ -291,7 +293,7 @@ export class MaterialRequestsService {
 			firstName: userInfo.firstName,
 			lastName: userInfo.lastName,
 		};
-		this.campaignMonitorService.sendForMaterialRequest(emailInfo);
+		await this.campaignMonitorService.sendForMaterialRequest(emailInfo);
```

Comment:
You can make this a little bit more efficient by sending all the emails at once.

```typescript
		// Send mail to each maintainer containing only material requests for objects they are the maintainer of
		const cpEmailSendRequests = groupedArray.map(async (materialRequests: MaterialRequest[]) => {
			const emailInfo: MaterialRequestEmailInfo = {
				// Each materialRequest in this group has the same maintainer, otherwise, the maintainer will receive multiple mails
				to: materialRequests[0].contactMail,
				isToMaintainer: true,
				template: Template.MATERIAL_REQUEST_MAINTAINER,
				materialRequests: materialRequests,
				sendRequestListDto,
				firstName: userInfo.firstName,
				lastName: userInfo.lastName,
			};
			await this.campaignMonitorService.sendForMaterialRequest(emailInfo);
		});
		
		// Send mail to the requester containing all of their material requests for all the objects they requested
		const emailInfo: MaterialRequestEmailInfo = {
			to: materialRequests[0].requesterMail,
			isToMaintainer: false,
			template: Template.MATERIAL_REQUEST_REQUESTER,
			materialRequests: materialRequests,
			sendRequestListDto,
			firstName: userInfo.firstName,
			lastName: userInfo.lastName,
		};
		const requesterEmailSendRequest = this.campaignMonitorService.sendForMaterialRequest(emailInfo);
		
		
		await Promise.all([...cpEmailSendRequests, requesterEmailSendRequest ]);
```